### ISSUE TITLE
[Feature Request]: Take mocha configuration from package.json

### ISSUE THREAD

**Issue by Enubia** [CoderLine/mocha-vscode#171]
### Description

Mocha allows you to specify the configuration inside the `package.json` file. It would be awesome if the extension could also read it from there since I don't want to add a configuration file to all my projects when the configuration is already provided inside the `package.json` file.

### Describe the solution you'd like

Provide the user with an option to read the config from the `package.json` instead of a needing a dedicated configuration file.

**Comment by Danielku15** [comment]
Technically it would be easy to support, but there is a reason why I didn't add it in first place:

* Currently the extension only needs to activate and load when there is really a mocha configuration file in the workspace. With supporting `package.json` it would mean the extension is loaded for almost every workspace. 
* While not really confirmed by numbers, I thought it is rather unusual to put too many framework configs in the package.config. It can easily bloat and things would be published for npm packages. The ecosystem somehow agreed and evolved to a direction that frameworks have their own dedicated configuration files to keep things isolated and benefit from scriptable configurations. 

But generally you are right, the extension should support it as Mocha supports it.

**Comment by Enubia** [comment]
> While not really confirmed by numbers, I thought it is rather unusual to put too many framework configs in the package.config. It can easily bloat and things would be published for npm packages. The ecosystem somehow agreed and evolved to a direction that frameworks have their own dedicated configuration files to keep things isolated and benefit from scriptable configurations.

I agree with you that it's usually the case that you have configuration files in place for all those things. But some projects (like the ones I have to work on a daily base) were configured when it wasn't commonly adapted to use those files and configurations were done in the package.json if possible.

In any case, thanks for this extension. It's the only one I found that properly works with typescript so keep up the good work. 👍
